odoo/models.py

```diff
@@ -612,7 +618,7 @@ class BaseModel(metaclass=MetaModel):
     """
     _table = None                   #: SQL table name used by model if :attr:`_auto`
     _table_query = None             #: SQL expression of the table's content (optional)
-    _sql_constraints: list[tuple[str, str, str]] = []   #: SQL constraints [(name, sql_def, message)]
+    _sql_objects: dict[str, SQLObject] = frozendict()  #: SQL constraints and indexes
```

Comment:
Because it is easier to manage one dict; these objects are managed in the same way: for example unique constraint and unique index are 2 objects, but they have the same purpose.